webapi/src/main/scala/org/knora/webapi/routing/v2/SearchRouteV2.scala

```diff
@@ -429,11 +429,12 @@ class SearchRouteV2(routeData: KnoraRouteData) extends KnoraRoute(routeData) wit
     "v2" / "searchbylabel" / Segment
   ) { searchval => // TODO: if a space is encoded as a "+", this is not converted back to a space
     get { requestContext =>
-      val searchString =
+      val sparqlEncodedSearchString =
         stringFormatter.toSparqlEncodedString(
           searchval,
           throw BadRequestException(s"Invalid search string: '$searchval'")
         )
```

Comment:
is this step still necessary if we replace some special characters anyways? I haven't checked - but I thought that just replaces some characters too

Reply:
`toSparqlEncodedString` returns an error when there is a new line or an empty string handed over. Also, the characters that are replaced are different than the ones that are replaced because of the Lucene Query Parser syntax. I thought it is better to leave them both separate - although there is some overlap (`\"` and `\\`).